t/plugin/opa2.t

```diff
@@ -0,0 +1,217 @@
+                            "disable": false,
+                            "key": "test-key"
+                        }
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: setup service
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/services/s1',
+                ngx.HTTP_PUT,
+                [[{
+                    "name": "s1",
+                    "plugins": {
+                        "key-auth": {
+                            "disable": false
+                        }
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 4: setup route with APISIX data
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "opa": {
+                            "host": "http://127.0.0.1:8181",
+                            "policy": "example2",
+                            "with_route": true,
+                            "with_upstream": true,
+                            "with_consumer": true,
+                            "with_service": true
+                        }
+                    },
+                    "upstream_id": "u1",
+                    "service_id": "s1",
+                    "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 5: hit route (test without apikey)
+--- request
+GET /hello
+--- more_headers
+test-header: only-for-test
+--- error_code: 401
+--- response_body eval
+qr/Missing API key found in request/
+
+
+
+=== TEST 6: hit route (test route data)
+--- request
+GET /hello
+--- more_headers
+test-header: only-for-test
+apikey: test-key
+--- error_code: 403
+--- response_body eval
+qr/\"route\":/ and qr/\"id\":\"r1\"/ and qr/\"plugins\":\{\"opa\"/ and
+qr/\"with_route\":true/
+
+
+
+=== TEST 7: hit route (test upstream data)
+--- request
+GET /hello
+--- more_headers
+test-header: only-for-test
+apikey: test-key
+--- error_code: 403
+--- response_body eval
+qr/\"upstream\":/ and qr/\"id\":\"u1\"/ and qr/\"nodes\":\[\{/ and
+qr/\"host\":\"127.0.0.1\"/ and qr/\"port\":1980/ and
+qr/\"weight\":1/ and qr/\"with_upstream\":true/ and
+qr/\"type\":\"roundrobin\"/
+
+
+
+=== TEST 8: hit route (test consumer data)
```

Comment:
Let's add test for no consumer/service cases

Reply:
added